src/drivers/uavcan_v1/ParamManager.hpp

```diff
@@ -63,14 +63,18 @@ class UavcanParamManager
 
 private:
 
-	const UavcanParamBinder _uavcan_params[8] {
-		{"uavcan.pub.esc.0.id",   "UCAN1_ESC_PUB"},
-		{"uavcan.pub.servo.0.id", "UCAN1_SERVO_PUB"},
-		{"uavcan.pub.gps.0.id",   "UCAN1_GPS_PUB"},
-		{"uavcan.sub.esc.0.id",   "UCAN1_ESC0_PID"},
-		{"uavcan.sub.gps.0.id",   "UCAN1_GPS0_PID"},
-		{"uavcan.sub.gps.1.id",   "UCAN1_GPS1_PID"},
-		{"uavcan.sub.bms.0.id",   "UCAN1_BMS0_PID"},
-		{"uavcan.sub.bms.1.id",   "UCAN1_BMS1_PID"},
+	const UavcanParamBinder _uavcan_params[10] {
+		{"uavcan.pub.esc.0.id",              "UCAN1_ESC_PUB"},
+		{"uavcan.pub.servo.0.id",            "UCAN1_SERVO_PUB"},
+		{"uavcan.pub.gps.0.id",              "UCAN1_GPS_PUB"},
+		{"uavcan.sub.esc.0.id",              "UCAN1_ESC0_PID"},
+		{"uavcan.sub.gps.0.id",              "UCAN1_GPS0_PID"},
+		{"uavcan.sub.gps.1.id",              "UCAN1_GPS1_PID"},
+		{"uavcan.sub.energy_source.0.id",      "UCAN1_BMS_ES_PID"},
+		{"uavcan.sub.battery_status.0.id",     "UCAN1_BMS_BS_PID"},
+		{"uavcan.sub.battery_parameters.0.id", "UCAN1_BMS_BP_PID"},
+		{"uavcan.sub.legacy_bms.0.id",         "UCAN1_LG_BMS_PID"},
+		//{"uavcan.sub.bms.0.id",   "UCAN1_BMS0_PID"}, //FIXME instancing
+		//{"uavcan.sub.bms.1.id",   "UCAN1_BMS1_PID"},
```

Comment:
@pavel-kirienko what do you mean by the instance qualifier vs. the prefix?  I think Peter and I both had a different interpretation of what that statement means.  Are you saying that instead of:
`uavcan.pub.esc.0.id`
it should be:
`uavcan.pub.0.esc.id`?

Reply:
What I mean is that instead of `uavcan.sub.battery_parameters.0.id` it should be `uavcan.sub.battery0.battery_parameters.id`. The convention does not apply to your specific example because there is no corresponding network service definition.

Reply:
So the formatting is like this?
uavcan.sub.<service_name><instance>.<subject_name>.id

Reply:
@PetervdPerk-NXP yes